### DISCUSSION TITLE
Do you support Google Analytics?

### DISCUSSION THREAD

**Discussion by dunithd** [thuliteio/doks#227] [Q&A] [answered]
I need to insert Google Analytics tracking code to Doks theme. I tried putting it inside the `config.toml` file. But it didn't appear to work. 

Any help is appreciated!

**Comment by h-enk** [comment, marked as answer]
Put your GA tracking code in `./layouts/partials/head/script-header.html`

Note, make sure to update the Content Security Policy in `./layouts/index.headers` to make it work. See also the Doks docs: [Content Security Policy](https://getdoks.org/docs/reference-guides/security/#content-security-policy)

**Reply by dunithd** [reply]
Thanks for the fast help. I added this snippet to `./layouts/partials/head/script-header.html` and stats started to appear on the GA console. 
However, I didn't understand what's inside `./layouts/index.headers` as I'm very new to Doks as well as Hugo. I didn't make changes to that file anyway.

```
<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=G-E0PQ5KH92S"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-XXXX');
</script>
```

**Reply by james-s-w-clark** [reply]
How should the Content Security Policy be updated for GA to work? I added the GA script to the suggested file `layouts\partials\head\script-header.html` and generated a new CSP with https://chrome.google.com/webstore/detail/content-security-policy-c/ahlnecfloencbkpfnpljbojmjkfgnmdc/related, but I don't see what I should plug into the CSP from this

**Reply by h-enk** [reply]
View your production website in e.g. Google Chrome, open developer tools (F!2). In the `Console` tab, you'll probably see some errors — these need to be fixed.

For instance,
![Snag_9d05b25](https://user-images.githubusercontent.com/3902872/165820645-91d8e5ef-a2d5-4189-9073-34977c7701ae.png)

Add to the `script-src` part of the `Content-Security-Policy` section in `.layouts/index.headers`:
1. `https://www.googletagmanager.com/gtag/js`
2. the `sha256-` part ( see example image above)
3. probably more — see your specific errors in the `Console` tab

**Reply by constantcopyandwork** [reply]
After adding GA, it would be necessary to add cookie warning to the website. I found a plugin by Osano <https://www.osano.com/cookieconsent/download/> that has an open source edition.
Has anyone implemented this or something similar?
Thanks!
